text/0000-cfg-oneof.md

```diff
@@ -0,0 +1,96 @@
+- Feature Name: `config_oneof`
+- Start Date: 2020-07-18
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Addition of `oneof` to existing `all`, `any`, `not` configuration predicates in `#[cfg!()]` macros
+
+# Motivation
+[motivation]: #motivation
+
+In a number of situations (particularly involving `no_std` and cross-target applications) it is useful to ensure that only one of a set of features are enabled.
+It is important to enforce this due to the additive behaviour of features (ie. dependencies may include the same sub-dependencie with different features enabled the result is a union of the enabled features), and difficult to specify for larger feature sets using existing predicates as the complexity for this increases exponentially with the number of exclusive features.
+
+The desired outcome of this is the ability to specify `#[cfg(oneof(feature="a", feature="b"))] do_something()` and `#[cfg(not(oneof(feature="a", feature="b")))] compile_error!(...)` to specify exclusive features without manually defining all possible valid/invalid combinations for the exclusive subset of features, and simplify other configurations that benefit from the `oneof` predicate.
+This allows authors of crates with exclusive feature sets to communicate this to consumers without requiring users to infer the feature issue from compiler errors.
```

Comment:
This isn't the *only* way to allow such a thing. As you mention yourself:

> without manually defining all...

Reply:
sure, though given the burden of the current approach crate authors _do not_ do this.